docs/integration_setup_failures.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Handling Setup Failures"
+---
+
+Your integration may not be able to be set up for a variety of reasons. The most common cases are because the device or service is offline or the credentials are no longer valid. Your integration must retry setup so it can recover as soon as reasonably possible when the device or service is back online without the user having to restart Home Assistant.
+
+## Handling offline or unavailable devices and services
+
+### Integrations using `async_setup_entry`
+
+Raise the `ConfigEntryNotReady` exception from `async_setup_entry` in the integration's `__init__.py`, and Home Assistant will automatically take care of retrying set up later. To avoid doubt, raising `ConfigEntryNotReady` in a platform's `async_setup_entry` is ineffective because it is too late to be caught by the config entry setup.
+
+#### Example
+
+```python
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Setup the config entry for my device."""
+    device = MyDevice(entry.data[CONF_HOST])
+    try:
+        await device.async_setup()
+    except (asyncio.TimeoutError, TimeoutException) as ex:
+        raise ConfigEntryNotReady(f"Timeout while connecting to {device.ipaddr}") from ex
+```
+
+If you are using a [DataUpdateCoordinator](integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities), calling `await coordinator.async_config_entry_first_refresh()` will also trigger this exception automaticlly if the first refresh failed.
+
+If your integration supports discovery, Home Assistant will automatically retry as soon as your device or service gets discovered.
+
+#### Handling logging of a retry
+
+Pass the error message to `ConfigEntryNotReady` as the first argument. Home Assistant will log the retry once with a log level of
+`warning`, and subsequent retires are logged at `debug` level. The error message will also be propagated to the UI and shown on the integrations page. Suppose you do not set a message when raising `ConfigEntryNotReady`; in that case, Home Assistant will try to extract the reason from the exception that is the cause of `ConfigEntryNotReady` if it was propagated from another exception.
```

Comment:
subsequent retries

Reply:
https://github.com/home-assistant/developers.home-assistant/pull/1363

I had to stare at that for a minute because I thought I spelled `subsequent` wrong, and then I realized it was `retries`.  😆